medusa/providers/torrent/json/hdbits.py

```diff
@@ -44,50 +33,74 @@ def __init__(self):
         self.url = 'https://hdbits.org'
         self.urls = {
             'search': urljoin(self.url, '/api/torrents'),
-            'rss': urljoin(self.url, '/api/torrents'),
-            'download': urljoin(self.url, '/download.php'),
+            'download': urljoin(self.url, '/download.php?{0}'),
         }
 
         # Proper Strings
+        self.proper_strings = ['PROPER', 'REPACK']
 
         # Miscellaneous Options
 
         # Torrent Stats
+        self.minseed = None
+        self.minleech = None
 
         # Cache
-        self.cache = HDBitsCache(self, min_time=15)  # only poll HDBits every 15 minutes max
+        self.cache = tv.Cache(self, min_time=10)  # Only poll HDBits every 10 minutes max
 
     def search(self, search_strings, age=0, ep_obj=None):
         """
         Search a provider and parse the results.
 
-        :param search_strings: A dict with mode (key) and the search value (value)
+        :param search_strings: A dict with {mode: search value}
         :param age: Not used
         :param ep_obj: Not used
         :returns: A list of search results (structure)
         """
-        # FIXME
         results = []
 
-        logger.log('Search string: {0}'.format(search_strings), logger.DEBUG)
+        log.debug('Search strings {0}', search_strings)
 
         self._check_auth()
 
-        response = self.get_url(self.urls['search'], post_data=search_strings, returns='response')
-        if not response or not response.content:
-            logger.log('No data returned from provider', logger.DEBUG)
-            return results
+        # Search Params
+        search_params = {
+            'username': self.username,
+            'passkey': self.passkey,
+            'category': [
+                # 1, # Movie
+                2, # TV
+                3, # Documentary
+                # 4, # Music
+                5, # Sport
+                # 6, # Audio Track
+                # 7, # XXX
+                # 8, # Misc/Demo
+            ],
+        }
+
+        for mode in search_strings:
+            log.debug('Search mode {0}', mode)
+
+            for search_string in search_strings[mode]:
 
-        if not self._check_auth_from_data(response):
-            return results
+                if mode != 'RSS':
+                    log.debug('Search string {search}', {'search': search_string})
+                    search_params['search'] = search_string
+                response = self.session.post(self.urls['search'], data=search_params)
```

Comment:
If it needs to be posted as json then instead of json.dumps use requests json param.